modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -0,0 +1,242 @@
+            LogicalTopologyService topologyService,
+            TablesConfiguration tablesConfiguration,
+            LeaseTracker leaseTracker,
+            HybridClock clock
+    ) {
+        this.msManager = msManager;
+        this.leaseTracker = leaseTracker;
+        this.clock = clock;
+
+        this.assignmentsTracker = new AssignmentsTracker(vaultManager, msManager, tablesConfiguration);
+        this.topologyTracker = new TopologyTracker(topologyService);
+        this.updater = new Updater();
+    }
+
+    /**
+     * Initializes the class.
+     */
+    public void init(String nodeName) {
+        this.nodeName = nodeName;
+
+        topologyTracker.startTrack();
+        assignmentsTracker.startTrack();
+    }
+
+    /**
+     * De-initializes the class.
+     */
+    public void deInit() {
+        topologyTracker.stopTrack();
+        assignmentsTracker.stopTrack();
+    }
+
+    /**
+     * Activates a lease updater to renew leases.
+     */
+    public void activate() {
+        //TODO: IGNITE-18879 Implement lease maintenance.
+        updaterTread = new Thread(updater, NamedThreadFactory.threadPrefix(nodeName, "lease-updater"));
+
+        updaterTread.start();
+    }
+
+    /**
+     * Runnable to update lease in Metastorage.
+     */
+    private class Updater implements Runnable {
+        @Override
+        public void run() {
+            while (!updaterTread.isInterrupted()) {
+                for (Map.Entry<ReplicationGroupId, Set<Assignment>> entry : assignmentsTracker.assignments().entrySet()) {
+                    ReplicationGroupId grpId = entry.getKey();
+
+                    Lease lease = leaseTracker.getLease(grpId);
+
+                    HybridTimestamp now = clock.now();
+
+                    // Nothing holds the lease.
+                    if (lease.getLeaseExpirationTime() == null
+                            // The lease is near to expiration.
+                            || now.getPhysical() > (lease.getLeaseExpirationTime().getPhysical() - LEASE_PERIOD / 2)) {
+                        var leaseKey = ByteArray.fromString(PLACEMENTDRIVER_PREFIX + grpId);
+
+                        var newTs = new HybridTimestamp(now.getPhysical() + LEASE_PERIOD, 0);
+
+                        ClusterNode candidate = nextLeaseHolder(entry.getValue());
+
+                        // Now nothing is holding the lease, and new lease is being granted.
+                        if (lease.getLeaseholder() == null && candidate != null
+                                // The lease is prolongation
+                                || lease.getLeaseholder() != null && lease.getLeaseholder().equals(candidate)
+                                // New lease is being granted, and the previous lease is expired.
+                                || candidate != null && (lease.getLeaseExpirationTime() == null
+                                || compareWithClockSkew(now, lease.getLeaseExpirationTime()) > 0)) {
```

Comment:
This condition is too complicated and the part with comparing with clock skew is a final nail in the coffin.
First, seems that the candidate should be not null in any case, why several comparisons?
Does anything actually depend on the current leaseholder?
Some parts, for example related to lease expiration time, could be assigned to separate variables.
And are you sure that we need to implement this right now instead of putting a todo with some of maintenance tickets?